src/openforms/formio/dynamic_config/__init__.py

```diff
@@ -44,7 +45,17 @@ def rewrite_formio_components_for_request(
     Loop over the formio configuration and inject request-specific configuration.
     """
     for component in configuration_wrapper:
-        register.update_config_for_request(component, request=request)
+        # prevent multiple overwrites of content components (bandaid for #2895)
+        if html_string := component.get("html", None):
+            soup = BeautifulSoup(html_string, "lxml")
+            tags = soup.find_all()
+            # find style attributes (not tags)
+            styles = [tag.get("style", None) for tag in tags]
+            nonce_missing = any(styles) and not all(
+                getattr(style, "nonce", None) for style in styles
+            )
+        if component["type"] != "content" or nonce_missing:
+            register.update_config_for_request(component, request=request)
```

Comment:
what happens now if I have a component with malicious HTML?

```html
<div>
  <style nonce="my-malicious-and-bad-nonce"></style>
  <script>alert('xss')</script>
</div>
```

If I read the code correctly, this means the style tag will be found, and all found style tags have a nonce (even though this is a spoofed nonce), so `nonce_missing` will be `False`, causing the component not to be post processed even though it should have been.

1. I'm not sure what `soup.find_all()` does without arguments - keep in mind that style tags can in theory be nested deeply
2. you need to compare against the server-side known value of the nonce, this is set by the CSP middleware on the request object. You can only leave the HTML untouched if all style tags have a nonce value that's equal to the one set on the request. This prevents malicious actors providing hardcoded nonces.

This is not the right place to do this sanitation - you should make this specific to the Content component: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/formio/components/vanilla.py#L173

Reply:
Wouldn't it be enough though for the current issue to make sure that there are no longer tags with inline styles? 
Since this is a bandaid trying to figure out if the function has already run or not, i think that checking whether the nonce are consistent is not really that necessary?

Reply:
> Wouldn't it be enough though for the current issue to make sure that there are no longer tags with inline styles?

No, while having run the code once does imply absence of inline styles, the absence of inline styles doesn't imply the code has already run before. I haven't checked what `register.update_config_for_request` does, but if I understand correctly it should be called at least (and preferably at most) once. Absence of inline styles does not guarantee that.

Reply:
but for the content component,  `register.update_config_for_request` calls https://github.com/open-formulieren/open-forms/blob/master/src/openforms/formio/components/vanilla.py#L173 (which, if I understand correctly, checks if there are inline styles and if they are present they are transformed in style tags with a nonce). If there are no inline styles, the `rewrite_for_request` for the content shouldn't run again. So I think that the check should be moved to the `rewrite_for_request` but then checking there if there are inline styles should be enough?